libr/util/strbuf.c

```diff
@@ -119,14 +121,32 @@ R_API bool r_strbuf_slice(RStrBuf *sb, int from, int len) {
 	return true;
 }
 
-R_API bool r_strbuf_set(RStrBuf *sb, const char *s) {
+R_API bool r_strbuf_setptr(RStrBuf *sb, char *s, int len) {
 	r_return_val_if_fail (sb, false);
+	if (len < 0) {
+		sb->len = strlen (s);
+		sb->ptrlen = sb->len + 1;
+	} else {
+		sb->ptrlen = len;
+		sb->len = len;
+	}
+	sb->ptr = s;
+	sb->weakref = true;
+	return true;
+}
 
+R_API bool r_strbuf_set(RStrBuf *sb, const char *s) {
+	r_return_val_if_fail (sb, false);
 	if (!s) {
 		r_strbuf_init (sb);
 		return true;
 	}
-	return r_strbuf_setbin (sb, (const ut8*)s, strlen (s));
+	size_t len = strlen (s);
+	if (!r_strbuf_setbin (sb, (const ut8*)s, len + 1)) {
```

Comment:
Why `len + 1`? 
`r_strbuf_setbin()` puts 0 at the end of buffer.
https://github.com/radareorg/radare2/blob/79fcf4fbff5cc8fb2711dd4c1e7f5521a43c531d/libr/util/strbuf.c#L90